### PR TITLE
ARCH-469 Redact specified headers

### PR DESCRIPTION
**Pull request by mcinbell**
## Description & Context
Introduces an option to proxay which will allow sensitive data in request headers to be redacted in the associated tapes. As we generally commit proxay tapes to source control this allows us to keep sensitive header values, such as access control keys and auth tokens, out of the tapes.

### CODE REVIEW

File: src/persistence.ts [hunk 1]
```diff
@@ -15,13 +15,15 @@ import {
  * Persistence layer to save tapes to disk and read them from disk.
  */
 export class Persistence {
-  constructor(private tapeDir: string) {}
+  constructor(private tapeDir: string, private redactHeaders: string[]) {}
```
Comment (fwouts) [inline src/persistence.ts:18]: Suggestion: These two could probably be readonly, might as well :)
```suggestion
  constructor(private readonly tapeDir: string, private readonly redactHeaders: string[]) {}
```

File: src/cli.ts [hunk 2]
```diff
@@ -5,6 +5,10 @@ import program from "commander";
 import fs from "fs-extra";
 import { RecordReplayServer } from "./server";
 
+function commaSeparatedList(value: string, _dummyPrevious?: any) {
```
Comment (fwouts) [inline src/cli.ts:8]: I think you should be able to omit the second parameter:
```suggestion
function commaSeparatedList(value: string) {
```

File: src/persistence.spec.ts [hunk 3]
```diff
@@ -77,6 +77,31 @@ const UTF8_RESPONSE_GZIP = gzipSync(Buffer.from(UTF8_RESPONSE, "utf8"));
...
+      response: {
+        status: {
+          code: 200,
+        },
+        headers: {},
+        body: Buffer.from(UTF8_RESPONSE, "utf8"),
+      },
+    };
+    redactRequestHeaders(record, ["x-auth-token"]);
+    expect(record.request.headers["x-auth-token"]).not.toEqual(sensitiveData);
```
Comment (fwouts) [inline src/persistence.spec.ts:100]: **Suggestion:** We could be more specific here?
```suggestion
    expect(record.request.headers["x-auth-token"]).toEqual("XXXX");
```

File: src/cli.ts (hunk 2 above)
Comment (mcinbell) [reply src/cli.ts:8]: Interesting, I was following the example and it included the `dummyPrevious` argument. Will try it out and see. Thanks.

**Pull request merged**